### PR TITLE
Add support for HTS precompiles

### LINKED ISSUE #2925: Support HTS precompiles
### Problem

[HIP-206](https://hips.hedera.com/hip/hip-206) is adding support for HTS precompiled contracts. When these are invoked from a contract call or contract create, they will generate additional child transactions in the record stream that the mirror node needs to handle

### Solution

- Update protobuf version
- Add new response codes
- Add `transaction.parent_consensus_timestamp` and populate it with `TransactionRecord.parent_consensus_timestamp`
- Add `transaction.nonce` with a default of 0 and populate it with `TransactionID.nonce`
- Support parsing new `Key.delegatable_contract_id`

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by steven-sheehy**
**Description**:
- Add new `transaction.nonce` and populate it
- Add new `transaction.parent_consensus_timestamp` and populate it
- Rename `scripts/timescaledb` to `scripts/v2`

**Related issue(s)**:

Fixes #2925 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: docs/operations.md [hunk 1]
```diff
@@ -141,12 +141,12 @@ For mirror node operators running the v1 database schema, the following steps ca
 4. Configure properties
 
    First, locate the directory that contains the TimescaleDB migration scripts. For source code, the target directory is
-   located at `hedera-mirror-importer/src/main/resources/db/scripts/timescaledb`. For binaries, the migration directory
+   located at `hedera-mirror-importer/src/main/resources/db/scripts/v2`. For binaries, the migration directory
```
Comment (Nana-EC) [inline docs/operations.md:144]: Q: Why just these 2 lines when the whole section referencing timescale.
We could just remove the section or update it to show it's in transition from Timescale to Citus

Also following these steps will no longer provide the desired outcome as the migration files under v2 are no longer applicable to these steps

**Review by Nana-EC — CHANGES_REQUESTED** [review]
I don't think you should add the timescaledb -> citus migration file moves?
`migration.config`, `migration.sh` are not applicable to citus and with the current changes it would make the documentation misleading as running migration files won't work or produce the original working intent.

There's a ticket that deals with this so we should leave it for that.

File: docs/operations.md (hunk 1 above)
Comment (steven-sheehy) [reply docs/operations.md:144]: I think you're missing the point. I wasn't converting the migrations from TimescaleDB to CitusDB. I was just renaming the scripts folder to be a more generic name that still works with both TimescaleDB and/or CitusDB. I'm also not adjusting the entire docs for CitusDB just making sure the references to the scripts path is correct.

Comment (Nana-EC) [reply docs/operations.md:144]: Understood, I get that you're renaming stuff and updating the docs that reference that path.
Just pointing out the result for someone without context may mislead them.
It's fine, the script migration flow is already broken so we'll address it and the v2 documentation later in the appropriate tickets.

File: hedera-mirror-rest/model/transaction.js [hunk 2]
```diff
@@ -61,6 +39,10 @@ class Transaction {
   static MEMO_FULL_NAME = `${this.tableAlias}.${this.MEMO}`;
   static NODE_ACCOUNT_ID = `node_account_id`;
   static NODE_ACCOUNT_ID_FULL_NAME = `${this.tableAlias}.${this.NODE_ACCOUNT_ID}`;
+  static NONCE = `nonce`;
+  static NONCE_FULL_NAME = `${this.tableAlias}.${this.NONCE}`;
+  static PARENT_CONSENSUS_TIMESTAMP = `parent_consensus_timestamp`;
+  static PARENT_CONSENSUS_TIMESTAMP_FULL_NAME = `${this.tableAlias}.${this.PARENT_CONSENSUS_TIMESTAMP}`;
```
Comment (Nana-EC) [inline hedera-mirror-rest/model/transaction.js:45]: nit: another option is to add the getFullName() we've started using.
However, this is consistent with current usage.

```javascript
  static getFullName(columnName) {
    return `${this.tableAlias}.${columnName}`;
  }
```

**Review by Nana-EC — APPROVED** [review]

**Review by ijungmann — COMMENTED** [review]
LGTM

**Review by ijungmann — APPROVED** [review]

**Review by xin-hedera — APPROVED** [review]
LGTM

**Pull request merged**